app/presenters/saml_request_presenter.rb

```diff
@@ -37,17 +38,33 @@ def requested_attributes
   attr_reader :request, :service_provider
 
   def ial2_authn_context?
-    (Saml::Idp::Constants::IAL2_AUTHN_CONTEXTS & authn_context).present?
+    ial_context.ial2_requested?
   end
 
   def ialmax_authn_context?
-    authn_context.include? Saml::Idp::Constants::IALMAX_AUTHN_CONTEXT_CLASSREF
+    ial_context.ialmax_requested?
   end
 
   def authn_context
     request.requested_authn_contexts
   end
 
+  def ial_context
+    @ial_context ||= IalContext.new(
+      ial: request.requested_ial_authn_context || default_ial_context,
+      service_provider: service_provider,
+      authn_context_comparison: request.requested_authn_context_comparison,
+    )
+  end
+
+  def default_ial_context
```

Comment:
Some duplication from the `SamlIdpAuthConcern` but not sure it's worth blowing everything up right now.